### DISCUSSION TITLE
SONOFF USB ZIGBEE DONGLE PLUS on SYNOLOGY DOCKER

### DISCUSSION THREAD

**Discussion by angeck** [Koenkk/zigbee2mqtt#11756] [Devices]
Hello,

I have a problem with this coordinator. I have Zigbee on a docker macvlan (Jeedom) but my new key is not online (before it's a CC2351) : 

```
lsusb
|__usb1          1d6b:0002:0404 09  2.00  480MBit/s 0mA 1IF  (Linux 4.4.180+ xhci-hcd xHCI Host Controller 0000:00:15.0) hub
  |__1-2         10c4:ea60:0100 00  2.00   12MBit/s 100mA 1IF  (ITead Sonoff Zigbee 3.0 USB Dongle Plus 5ebff58cb719ec11b20d76e5f01c6278)
  |__1-4         f400:f400:0100 00  2.00  480MBit/s 200mA 1IF  (Synology DiskStation 7F00126F381A2E78)
|__usb2          1d6b:0003:0404 09  3.00 5000MBit/s 0mA 1IF  (Linux 4.4.180+ xhci-hcd xHCI Host Controller 0000:00:15.0) hub
```

but

```
dmesg |grep tty
[    0.000000] Command line: root=/dev/md0 intel_iommu=igfx_off netif_num=2 HddEnableDynamicPower=1 syno_hw_version=DS220+ vender_format_version=2 synoboot2 sn=2140RLRJH4BRV macs=003332eb82d9,001132eb82da syno_ttyS0=pciserial,0x0:0x18.0x2,115200 syno_ttyS1=pciserial,0x0:0x18.0x0,115200
[    0.000000] Kernel command line: root=/dev/md0 intel_iommu=igfx_off netif_num=2 HddEnableDynamicPower=1 syno_hw_version=DS220+ vender_format_version=2 synoboot2 sn=2140RLRJH4BRV macs=003332eb82d9,001132eb82da syno_ttyS0=pciserial,0x0:0x18.0x2,115200 syno_ttyS1=pciserial,0x0:0x18.0x0,115200
[    4.506485] dw-apb-uart.1: ttyS1 at MMIO 0xa1215000 (irq = 54, base_baud = 115200) is a 16550A
[    4.522931] console [ttyS0] disabled
[    4.526772] dw-apb-uart.2: ttyS0 at MMIO 0xa1217000 (irq = 6, base_baud = 115200) is a 16550A
[    4.536458] console [ttyS0] enabled
[   27.752529] synobios open /dev/ttyS1 success
[   52.540966] systemd[1]: Created slice system-serial\x2dgetty.slice.
[   52.554092] systemd[1]: Starting system-serial\x2dgetty.slice.
[   55.487849] synobios open /dev/ttyS1 success
```

I have read the tuto from : https://www.geeek.org/synology-dsm-7-probleme-usb/ but the same, the docker don't see my key.

Thanks for your help and sorry for my English

**Comment by tuanha2000vn** [comment]
I have the same issue.

**Reply by 7uAip6** [reply]
Same here.  It seems I wasted my $ on this.

**Comment by sibalzer** [comment]
I found a solution. The problem is that the SONOFF Zigbee 3.0 USB Dongle Plus uses a different usb-to-serial chip, which driver is not present in DSM7. So you have to manualy install the kernel modul.

1. Get the cp210x.ko modul. E.g. precompiled from [here](https://github.com/robertklep/dsm7-usb-serial-drivers) ([you can find your platform here](https://kb.synology.com/en-global/DSM/tutorial/What_kind_of_CPU_does_my_NAS_have))
2. Copy the cp210x.ko file to /lib/modules/
3. Add `insmod /lib/modules/cp210x.ko` to the boot script [like here](https://mariushosting.com/synology-how-to-add-usb-support-on-dsm-7/)
4. Edit the docker-compose.yaml and change `/dev/ttyACM0:/dev/ttyACM0` to `/dev/ttyUSB0:/dev/ttyUSB0`
5. Change the serial port to `/dev/ttyUSB0` in the configuration.yaml

**Reply by lukasadc** [reply]
Where can I find docker-compose and configuration.yaml ?

**Reply by Jasonb11** [reply]
I just got this setup with Portainer on my synology. In portainer if you edit/duplicate the container you can scroll down to advanced container settings and then runtime and resources. Add device and choose the /dev/TTYUSB0 from the host and point it to the destination container using the same path /dev/TTYUSB0.

**Comment by Arelav** [comment]
I just want to say thank you. The driver disappeared after DSM update and as the previous time when I bough Synology it works like a charm.

**Comment by ltoinel** [comment]
@angeck  : I update my blog post on Geeek.org, the drivers path changed on the GitHub repository since the DSM 7.1 update !

**Comment by mariusrga** [comment]
Hi,

I have similar problem as well, a synology DS 216+ (braswell architecture) and, I used the drivers from here https://github.com/robertklep/dsm7-usb-serial-drivers/tree/main/modules/braswell/dsm-7.1 . Problem is that, when trying to use the insmod command, I get "insmod: ERROR: could not insert module /lib/modules/cp210x.ko: Invalid module format"

What am I doing wrong?

**Reply by cgarciaq** [reply]
I was getting this error message when I tried with a cp210x.ko compiled for another architecture. But I see DS216x is braswell... could it be that your disktation is another version of DS216?

**Reply by mariusrga** [reply]
Thanks for the reply. I was using wget to download directly the files, but seems that github.com was having another sublink underneath the files names, so once I got the correct files, all went as it should.

**Comment by RunnicFusion** [comment]
Hi! do you guys see the dongle when entering lsusb inside the teminal? After flashing the device isn't listed as sonoff any longer. Right now i get some hex like fffffdffff as name. And the device isn't recognized any longer. Do I need different drivers on my Synology than cp210x?

**Comment by madcowGit** [comment]
In case someone is still looking for this. Follow this guide:
https://github.com/robertklep/dsm7-usb-serial-drivers
